review/get_review_ids.py

```diff
@@ -89,19 +93,31 @@ def get_vertical(current_course):
     vertical_data = set()
 
     for block_key, state in get_records(user, current_course):
-        if is_valid_problem(store, block_key, state):
-            parent = course_blocks.get_parents(block_key)[0]
-            vertical_id = parent.block_id
-            vertical_data.add(vertical_id)
+        block_key = block_key.replace(course_key=store.fill_in_run(block_key.course_key))
+        if is_valid_problem(store, block_key, state, course_blocks):
+            vertical = course_blocks.get_parents(block_key)[0]
+            sequential = course_blocks.get_parents(vertical)[0]
+            # This is in case the direct parent of a problem is not a vertical,
+            # we want to keep looking until we find the parent vertical to display.
+            # For example, you may see:
+            # sequential -> vertical -> split_test -> problem
+            # OR
+            # sequential -> vertical -> vertical -> problem
+            # OR
+            # sequential -> vertical -> conditional_block -> problem
+            while sequential.block_type != 'sequential':
+                vertical = sequential
+                sequential = course_blocks.get_parents(vertical)[0]
```

Comment:
This will result in an `IndexError` if the problem does not have a `sequential` grandparent.

Reply:
Is that a possibility? I was assuming they would have to have a sequential parent due to the tree structure of a course. (My idea of how a course looks: 
course -> chapter -> sequential -> vertical -> problem.
Understanding that other weird components can be in the middle of that, I believed you had to still have all of these /essential/ components in order for it to display in the course and thus be a possibility of being shown)

Reply:
We have courses that don't follow that structure, especially XML-edited MIT courses.